src/main/kotlin/lotto/domain/LottoProvider.kt

```diff
@@ -1,10 +1,34 @@
 package lotto.domain
 
-class LottoProvider(payment: Int) {
-    val numberOfLottos: Int = payment / LOTTO_PRICE
-    val lottos: List<Lotto> = (1..numberOfLottos).map { Lotto() }
+import lotto.domain.numbers.LottoNumbers
+
+class LottoProvider(payment: Int, customLottoNumbers: List<LottoNumbers> = emptyList()) {
+    val numberOfCustomLottos: Int
+    val numberOfAutomaticLottos: Int
```

Comment:
컴파일러가 정의와 선언을 한번에 하는것을 추천하고 있네요~ 젯브레인에서 만든 언어라서 젯브레인 컴파일러인 인텔리제이가 추천하는대로 따라주는것도 좋습니다~!
```suggestion
        val numberOfCustomLottos: Int = when (isAffordable(payment, customLottoNumbers.size)) {
        true -> customLottoNumbers.size
        false -> throw IllegalArgumentException(TOO_MANY_LOTTO_REQUESTED)
    }
    val numberOfAutomaticLottos: Int = payment.getNumberOfAffordableLotto() - numberOfCustomLottos
```

Reply:
[622623c](https://github.com/next-step/kotlin-lotto/pull/386/commits/622623ca0d5ca6b92cb8d9b56eaf189f9a934f92) 에 반영했습니다!